utils/amdvlk_build_for_tag.py

```diff
@@ -0,0 +1,251 @@
+        os.chdir(self.srcDir)
+
+        repoName = 'AMDVLK'
+        initCmd='repo init -u ' + self.targetRepo + repoName + ' -b refs/tags/' + self.buildTag
+        syncCmd='repo sync -j8'
+        if self.IsBuildTagNewer():
+            initCmd += ' -m build_with_tools.xml'
+        else:
+            initCmd += ' -m default.xml'
+        if os.system(initCmd):
+            eprint(initCmd + ' failed')
+            exit(-1)
+        if os.system(syncCmd):
+            eprint('repo sync failed')
+            exit(-1)
+
+        # Simply use repo command instead of reading from manifest to get path
+        amdvlkPath = os.popen('repo list --path-only ' + repoName).read().strip()
+        self.driverRoot = self.srcDir + amdvlkPath[:-len(repoName)]
+        repo = git.Repo(amdvlkPath)
+        for tagRef in repo.tags:
+            if self.buildTag == tagRef.name:
+                if self.IsBuildTagNewer():
+                    self.descript = tagRef.tag.message
+                else:
+                    self.descript = tagRef.commit.message
+                self.version = self.buildTag[2:]
+                break
+
+        repo.git.checkout(self.buildTag)
+
+    def GenerateReleaseNotes(self):
+        os.chdir(self.workDir)
+        releaseNotes = '[Driver installation instruction](' + self.targetRepo + 'AMDVLK#install-with-pre-built-driver) \n\n'
+        formated_str = self.descript.replace('New feature and improvement', '## New feature and improvement')
+        formated_str = formated_str.replace('Issue fix', '## Issue fix')
+        releaseNotes += formated_str
+        with open('amdvlk_releaseNotes.md', 'w') as notes:
+            notes.write(releaseNotes + '\n')
+
+    def Build(self):
+        self.PrepareChangelog()
+
+        if self.distro == 'Ubuntu':
+            self.MakeDriverPackage('64')
+            self.ArchiveAmdllpcTools('amd64')
+            self.MakeDriverPackage('32')
+            self.ArchiveAmdllpcTools('i386')
+            self.GenerateReleaseNotes()
+        elif self.distro == 'RHEL':
+            self.MakeDriverPackage('64')
+        print('The package is generated successfully for ' + self.buildTag)
+
+    def MakeDriverPackage(self, arch):
+        cmakeName = 'cmake '
+        if (self.distro == 'RHEL'):
+            cmakeName = 'source scl_source enable devtoolset-7 && cmake '
+
+        if not self.IsBuildTagNewer():
+            # Fetch spvgen resources
+            os.chdir(self.driverRoot + 'spvgen/external')
+            print(self.driverRoot + 'spvgen/external')
+            if os.system('python3 fetch_external_sources.py'):
+                eprint('SPVGEN: fetch external sources failed')
+                exit(-1)
+
+        self.buildDir   = 'xgl/Release64' if arch == '64' else 'xgl/Release32'
+        cmakeFlags = ' -G Ninja -S xgl -B ' + self.buildDir + ' -DBUILD_WAYLAND_SUPPORT=ON -DPACKAGE_VERSION=' + self.version + ' -DXGL_BUILD_TOOLS=ON'
```

Comment:
Just for your info, `-DBUILD_WAYLAND_SUPPORT=ON` is the default for a while, so it doesn’t need to be explicitly enabled.